inference/core/workflows/core_steps/common/utils.py

```diff
@@ -48,38 +56,80 @@ def attach_prediction_type_info(
     return predictions
 
 
-def attach_parent_info(
-    images: List[Dict[str, Any]],
-    predictions: List[Dict[str, Any]],
-    nested_key: Optional[str] = "predictions",
-) -> List[Dict[str, Any]]:
-    return [
-        attach_parent_info_to_prediction(
-            image=image, prediction=prediction, nested_key=nested_key
+def convert_to_sv_detections(
+    predictions: List[Dict[str, Union[List[Dict[str, Any]], Any]]],
+    predictions_key: str = "predictions",
+) -> List[sv.Detections]:
+    batch_of_detections: List[sv.Detections] = []
+    for p in predictions:
+        detections = sv.Detections.from_inference(p)
+        parent_ids = [d.get(PARENT_ID_KEY, "") for d in p[predictions_key]]
+        detection_ids = [
+            d.get(DETECTION_ID_KEY, str(uuid.uuid4)) for d in p[predictions_key]
+        ]
+        detections[DETECTION_ID_KEY] = np.array(detection_ids)
+        detections[PARENT_ID_KEY] = np.array(parent_ids)
+        batch_of_detections.append(detections)
+    return batch_of_detections
+
+
+def add_keypoints_to_detections(
+    predictions: List[Dict[str, Union[List[Dict[str, Any]], Any]]],
+    detections: sv.Detections,
+):
+    keypoints_class_names = []
+    keypoints_class_ids = []
+    keypoints_confidences = []
+    keypoints_xy = []
+    for p in predictions:
+        keypoints = p.get(KEYPOINTS_KEY, [])
+        keypoints_class_names.append(
+            np.array([k.get(KEYPOINTS_CLASS_NAME_KEY, "") for k in keypoints])
```

Comment:
[2] I would not make inner keypoints key optional - we shall assume them to be present or fail, as lack of them indicates bigger issue with contract